### PR TITLE
planner: make converge index merge path feel the prefer tiflash hint

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/56217

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/planner/core/find_best_task.go [hunk 1]
```diff
@@ -1356,6 +1356,10 @@ func findBestTask4DS(ds *DataSource, prop *property.PhysicalProperty, planCounte
 	for _, candidate := range candidates {
 		path := candidate.path
 		if path.PartialIndexPaths != nil {
+			// prefer tiflash, while current table path is tikv, skip it.
+			if ds.PreferStoreType&h.PreferTiFlash != 0 && path.StoreType == kv.TiKV {
```
Comment (AilinKid) [inline pkg/planner/core/find_best_task.go:1360]: when we step into here, we can ensure that
* tiflash is existed (replica set is checked)
* whether tiflash path can be generated successfully or not, not sure here, just keep the same logic from L1498

future work: we can sort the candidate paths according to the user's preferred hints or variables, then gen them first, if we do, output their preferred path, on the condition that the preferred path can be generated successfully. IF failed, then try next, inferior paths

**Review by elsa0520 — APPROVED** [review]
LGTM

**Review by qw4990 — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**